### PR TITLE
[MB-1576] Add lookup for DistanceZipSITOriginLookup

### PR DESCRIPTION
**Pull request by mboren**
## Description

This PR adds the lookup for the DistanceZipSITOrigin param key.

This is extremely similar to the lookup for `DistanceZipSITDest`, which was added in this PR: https://github.com/transcom/mymove/pull/5733

Since most of the code for the actual distance lookup is identical to the `DistanceZipSITDest` lookup, I thought about pulling it out into a helper function that could be used in both, but I figure we're more likely to want to change one and not the other, than we are to add a third lookup with the same logic.

## Acceptance testing
We can't test this directly on staging yet, so no acceptance is needed.

## Setup

```sh
make server_test
```

## Code Review Verification Steps
* [ ] If the change is risky, it has been tested in experimental before merging.
* [ ] Code follows the guidelines for [Logging](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#logging)
* [ ] The requirements listed in
 [Querying the Database Safely](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#querying-the-database-safely)
 have been satisfied.
* Any new migrations/schema changes:
  * [ ] Follow our guidelines for zero-downtime deploys (see [Zero-Downtime Deploys](https://github.com/transcom/mymove/wiki/migrate-the-database#zero-downtime-migrations))
  * [ ] Have been communicated to #g-database
  * [ ] Secure migrations have been tested following the instructions in our [docs](https://github.com/transcom/mymove/wiki/migrate-the-database#secure-migrations)
* [ ] There are no aXe warnings for UI.
* [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/master/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
* [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
* [ ] User facing changes have been reviewed by design.
* [ ] Request review from a member of a different team.
* [ ] Have the Jira acceptance criteria been met for this change?

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-1576) for this change

### CODE REVIEW

File: pkg/payment_request/service_param_value_lookups/service_param_value_lookups.go [hunk 1]
```diff
@@ -331,6 +345,14 @@ func ServiceParamLookupInitialize(
 		return nil, err
 	}
 
+	paramKey = models.ServiceItemParamNameDistanceZipSITOrigin
+	err = s.setLookup(serviceItemCode, paramKey, DistanceZipSITOriginLookup{
+		OriginalAddress: pickupAddress,
```
Comment (jacquelineIO) [inline pkg/payment_request/service_param_value_lookups/service_param_value_lookups.go:350]: @mboren  your `OriginalAddress` should be coming from `MTOServiceItem.SITOriginHHGOriginalAddress`.

File: pkg/payment_request/service_param_value_lookups/distance_zip_sit_origin_lookup_test.go [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+
+	mtoServiceItemDiffZip3 := testdatagen.MakeMTOServiceItem(suite.DB(),
+		testdatagen.Assertions{
+			PickupAddress: originAddress,
+			ReService:     reService,
+			Move:          move,
+			MTOShipment:   mtoShipment,
+			MTOServiceItem: models.MTOServiceItem{
+				SITOriginHHGActualAddressID: &actualOriginDiffZip3Address.ID,
+				SITOriginHHGActualAddress:   &actualOriginDiffZip3Address,
```
Comment (jacquelineIO) [inline pkg/payment_request/service_param_value_lookups/distance_zip_sit_origin_lookup_test.go:84]: You also need to set `MTOServiceItem.SITOriginHHGOriginalAddress`. You won't be using the address from the Shipment.

**Review by jacquelineIO — CHANGES_REQUESTED** [review]
SIT origin service items will have both `SITOriginHHGActualAddress` and `SITOriginHHGOriginalAddress` set. Since linehaul hasn't started yet, we can't ensure that the shipment won't be updated again before the linehaul happens. So we are keeping the information needed on the SIT origin service item. This is different than SIT destination, because linehaul has been completed, and we shouldn't be changing/update the shipment and we are able to use the destination address for distance.

File: pkg/payment_request/service_param_value_lookups/service_param_value_lookups.go (hunk 1 above)
Comment (mboren) [reply pkg/payment_request/service_param_value_lookups/service_param_value_lookups.go]: Ok, I'll do that

**Review by jacquelineIO — COMMENTED** [review]
The code looks good, but I think there is confusion on when to use the case statement at the beginning of the file. I left a clarifying comment on the code, to be considered for updating the code.

File: pkg/payment_request/service_param_value_lookups/service_param_value_lookups.go [hunk 3]
```diff
@@ -101,8 +106,27 @@ func ServiceParamLookupInitialize(
 			}
 			sitDestinationFinalAddress = *mtoServiceItem.SITDestinationFinalAddress
 		}
-		fallthrough
-	default:
```
Comment (jacquelineIO) [inline pkg/payment_request/service_param_value_lookups/service_param_value_lookups.go:105]: So getting rid of this `fallthrough` is why were having to use this new variable `	savePickupAndDestinationFromShipment := true`.

Comment (mboren) [reply pkg/payment_request/service_param_value_lookups/service_param_value_lookups.go]: Yeah. The boolean feels a bit awkward, but I wasn't sure how else to do this.

File: pkg/payment_request/service_param_value_lookups/service_param_value_lookups.go [hunk 4]
```diff
@@ -101,8 +106,27 @@ func ServiceParamLookupInitialize(
...
+	case models.ReServiceCodeDOASIT, models.ReServiceCodeDOFSIT, models.ReServiceCodeDOPSIT:
+		// load original origin address from service item
+		if mtoServiceItem.SITOriginHHGOriginalAddressID != nil && *mtoServiceItem.SITOriginHHGOriginalAddressID != uuid.Nil {
+			err = db.Load(&mtoServiceItem, "SITOriginHHGOriginalAddress")
+			if err != nil {
+				return nil, err
+			}
+			sitOriginOriginalAddress = *mtoServiceItem.SITOriginHHGOriginalAddress
```
[the remark this answers is not in the document]
Comment (mboren) [reply pkg/payment_request/service_param_value_lookups/service_param_value_lookups.go]: I took out the case that I added, removed the top level variables for SIT addresses, and moved the address loading into the `default` case. This feels a bit more sensible, but I'm still not sure i'm doing the right thing

**Comment by mboren** [comment]
Just realized that I do have a DB connection available in the lookup, so I'll be changing this a bit to use that later today

**Review by jacquelineIO — APPROVED** [review]
lgtm

**Pull request merged**